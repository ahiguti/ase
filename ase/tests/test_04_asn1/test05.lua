
local Test = ASE.CPP['ASETestCPP']
local x = ""
for i = 1, 1000 do
  x = x .. "a"
  local s = Test.ToDER(x)
  local v = Test.DERLen(s)
  if (not v[0]) then
    print("failed")
  else
    if (v[2]) then
      print("ov")
    else
      print("len=" .. v[1] .. " derlen=" .. #s)
    end
  end
end
