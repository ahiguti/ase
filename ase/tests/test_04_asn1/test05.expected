len=3 derlen=3
len=4 derlen=4
len=5 derlen=5
len=6 derlen=6
len=7 derlen=7
len=8 derlen=8
len=9 derlen=9
len=10 derlen=10
len=11 derlen=11
len=12 derlen=12
len=13 derlen=13
len=14 derlen=14
len=15 derlen=15
len=16 derlen=16
len=17 derlen=17
len=18 derlen=18
len=19 derlen=19
len=20 derlen=20
len=21 derlen=21
len=22 derlen=22
len=23 derlen=23
len=24 derlen=24
len=25 derlen=25
len=26 derlen=26
len=27 derlen=27
len=28 derlen=28
len=29 derlen=29
len=30 derlen=30
len=31 derlen=31
len=32 derlen=32
len=33 derlen=33
len=34 derlen=34
len=35 derlen=35
len=36 derlen=36
len=37 derlen=37
len=38 derlen=38
len=39 derlen=39
len=40 derlen=40
len=41 derlen=41
len=42 derlen=42
len=43 derlen=43
len=44 derlen=44
len=45 derlen=45
len=46 derlen=46
len=47 derlen=47
len=48 derlen=48
len=49 derlen=49
len=50 derlen=50
len=51 derlen=51
len=52 derlen=52
len=53 derlen=53
len=54 derlen=54
len=55 derlen=55
len=56 derlen=56
len=57 derlen=57
len=58 derlen=58
len=59 derlen=59
len=60 derlen=60
len=61 derlen=61
len=62 derlen=62
len=63 derlen=63
len=64 derlen=64
len=65 derlen=65
len=66 derlen=66
len=67 derlen=67
len=68 derlen=68
len=69 derlen=69
len=70 derlen=70
len=71 derlen=71
len=72 derlen=72
len=73 derlen=73
len=74 derlen=74
len=75 derlen=75
len=76 derlen=76
len=77 derlen=77
len=78 derlen=78
len=79 derlen=79
len=80 derlen=80
len=81 derlen=81
len=82 derlen=82
len=83 derlen=83
len=84 derlen=84
len=85 derlen=85
len=86 derlen=86
len=87 derlen=87
len=88 derlen=88
len=89 derlen=89
len=90 derlen=90
len=91 derlen=91
len=92 derlen=92
len=93 derlen=93
len=94 derlen=94
len=95 derlen=95
len=96 derlen=96
len=97 derlen=97
len=98 derlen=98
len=99 derlen=99
len=100 derlen=100
len=101 derlen=101
len=102 derlen=102
len=103 derlen=103
len=104 derlen=104
len=105 derlen=105
len=106 derlen=106
len=107 derlen=107
len=108 derlen=108
len=109 derlen=109
len=110 derlen=110
len=111 derlen=111
len=112 derlen=112
len=113 derlen=113
len=114 derlen=114
len=115 derlen=115
len=116 derlen=116
len=117 derlen=117
len=118 derlen=118
len=119 derlen=119
len=120 derlen=120
len=121 derlen=121
len=122 derlen=122
len=123 derlen=123
len=124 derlen=124
len=125 derlen=125
len=126 derlen=126
len=127 derlen=127
len=128 derlen=128
len=129 derlen=129
len=131 derlen=131
len=132 derlen=132
len=133 derlen=133
len=134 derlen=134
len=135 derlen=135
len=136 derlen=136
len=137 derlen=137
len=138 derlen=138
len=139 derlen=139
len=140 derlen=140
len=141 derlen=141
len=142 derlen=142
len=143 derlen=143
len=144 derlen=144
len=145 derlen=145
len=146 derlen=146
len=147 derlen=147
len=148 derlen=148
len=149 derlen=149
len=150 derlen=150
len=151 derlen=151
len=152 derlen=152
len=153 derlen=153
len=154 derlen=154
len=155 derlen=155
len=156 derlen=156
len=157 derlen=157
len=158 derlen=158
len=159 derlen=159
len=160 derlen=160
len=161 derlen=161
len=162 derlen=162
len=163 derlen=163
len=164 derlen=164
len=165 derlen=165
len=166 derlen=166
len=167 derlen=167
len=168 derlen=168
len=169 derlen=169
len=170 derlen=170
len=171 derlen=171
len=172 derlen=172
len=173 derlen=173
len=174 derlen=174
len=175 derlen=175
len=176 derlen=176
len=177 derlen=177
len=178 derlen=178
len=179 derlen=179
len=180 derlen=180
len=181 derlen=181
len=182 derlen=182
len=183 derlen=183
len=184 derlen=184
len=185 derlen=185
len=186 derlen=186
len=187 derlen=187
len=188 derlen=188
len=189 derlen=189
len=190 derlen=190
len=191 derlen=191
len=192 derlen=192
len=193 derlen=193
len=194 derlen=194
len=195 derlen=195
len=196 derlen=196
len=197 derlen=197
len=198 derlen=198
len=199 derlen=199
len=200 derlen=200
len=201 derlen=201
len=202 derlen=202
len=203 derlen=203
len=204 derlen=204
len=205 derlen=205
len=206 derlen=206
len=207 derlen=207
len=208 derlen=208
len=209 derlen=209
len=210 derlen=210
len=211 derlen=211
len=212 derlen=212
len=213 derlen=213
len=214 derlen=214
len=215 derlen=215
len=216 derlen=216
len=217 derlen=217
len=218 derlen=218
len=219 derlen=219
len=220 derlen=220
len=221 derlen=221
len=222 derlen=222
len=223 derlen=223
len=224 derlen=224
len=225 derlen=225
len=226 derlen=226
len=227 derlen=227
len=228 derlen=228
len=229 derlen=229
len=230 derlen=230
len=231 derlen=231
len=232 derlen=232
len=233 derlen=233
len=234 derlen=234
len=235 derlen=235
len=236 derlen=236
len=237 derlen=237
len=238 derlen=238
len=239 derlen=239
len=240 derlen=240
len=241 derlen=241
len=242 derlen=242
len=243 derlen=243
len=244 derlen=244
len=245 derlen=245
len=246 derlen=246
len=247 derlen=247
len=248 derlen=248
len=249 derlen=249
len=250 derlen=250
len=251 derlen=251
len=252 derlen=252
len=253 derlen=253
len=254 derlen=254
len=255 derlen=255
len=256 derlen=256
len=257 derlen=257
len=258 derlen=258
len=260 derlen=260
len=261 derlen=261
len=262 derlen=262
len=263 derlen=263
len=264 derlen=264
len=265 derlen=265
len=266 derlen=266
len=267 derlen=267
len=268 derlen=268
len=269 derlen=269
len=270 derlen=270
len=271 derlen=271
len=272 derlen=272
len=273 derlen=273
len=274 derlen=274
len=275 derlen=275
len=276 derlen=276
len=277 derlen=277
len=278 derlen=278
len=279 derlen=279
len=280 derlen=280
len=281 derlen=281
len=282 derlen=282
len=283 derlen=283
len=284 derlen=284
len=285 derlen=285
len=286 derlen=286
len=287 derlen=287
len=288 derlen=288
len=289 derlen=289
len=290 derlen=290
len=291 derlen=291
len=292 derlen=292
len=293 derlen=293
len=294 derlen=294
len=295 derlen=295
len=296 derlen=296
len=297 derlen=297
len=298 derlen=298
len=299 derlen=299
len=300 derlen=300
len=301 derlen=301
len=302 derlen=302
len=303 derlen=303
len=304 derlen=304
len=305 derlen=305
len=306 derlen=306
len=307 derlen=307
len=308 derlen=308
len=309 derlen=309
len=310 derlen=310
len=311 derlen=311
len=312 derlen=312
len=313 derlen=313
len=314 derlen=314
len=315 derlen=315
len=316 derlen=316
len=317 derlen=317
len=318 derlen=318
len=319 derlen=319
len=320 derlen=320
len=321 derlen=321
len=322 derlen=322
len=323 derlen=323
len=324 derlen=324
len=325 derlen=325
len=326 derlen=326
len=327 derlen=327
len=328 derlen=328
len=329 derlen=329
len=330 derlen=330
len=331 derlen=331
len=332 derlen=332
len=333 derlen=333
len=334 derlen=334
len=335 derlen=335
len=336 derlen=336
len=337 derlen=337
len=338 derlen=338
len=339 derlen=339
len=340 derlen=340
len=341 derlen=341
len=342 derlen=342
len=343 derlen=343
len=344 derlen=344
len=345 derlen=345
len=346 derlen=346
len=347 derlen=347
len=348 derlen=348
len=349 derlen=349
len=350 derlen=350
len=351 derlen=351
len=352 derlen=352
len=353 derlen=353
len=354 derlen=354
len=355 derlen=355
len=356 derlen=356
len=357 derlen=357
len=358 derlen=358
len=359 derlen=359
len=360 derlen=360
len=361 derlen=361
len=362 derlen=362
len=363 derlen=363
len=364 derlen=364
len=365 derlen=365
len=366 derlen=366
len=367 derlen=367
len=368 derlen=368
len=369 derlen=369
len=370 derlen=370
len=371 derlen=371
len=372 derlen=372
len=373 derlen=373
len=374 derlen=374
len=375 derlen=375
len=376 derlen=376
len=377 derlen=377
len=378 derlen=378
len=379 derlen=379
len=380 derlen=380
len=381 derlen=381
len=382 derlen=382
len=383 derlen=383
len=384 derlen=384
len=385 derlen=385
len=386 derlen=386
len=387 derlen=387
len=388 derlen=388
len=389 derlen=389
len=390 derlen=390
len=391 derlen=391
len=392 derlen=392
len=393 derlen=393
len=394 derlen=394
len=395 derlen=395
len=396 derlen=396
len=397 derlen=397
len=398 derlen=398
len=399 derlen=399
len=400 derlen=400
len=401 derlen=401
len=402 derlen=402
len=403 derlen=403
len=404 derlen=404
len=405 derlen=405
len=406 derlen=406
len=407 derlen=407
len=408 derlen=408
len=409 derlen=409
len=410 derlen=410
len=411 derlen=411
len=412 derlen=412
len=413 derlen=413
len=414 derlen=414
len=415 derlen=415
len=416 derlen=416
len=417 derlen=417
len=418 derlen=418
len=419 derlen=419
len=420 derlen=420
len=421 derlen=421
len=422 derlen=422
len=423 derlen=423
len=424 derlen=424
len=425 derlen=425
len=426 derlen=426
len=427 derlen=427
len=428 derlen=428
len=429 derlen=429
len=430 derlen=430
len=431 derlen=431
len=432 derlen=432
len=433 derlen=433
len=434 derlen=434
len=435 derlen=435
len=436 derlen=436
len=437 derlen=437
len=438 derlen=438
len=439 derlen=439
len=440 derlen=440
len=441 derlen=441
len=442 derlen=442
len=443 derlen=443
len=444 derlen=444
len=445 derlen=445
len=446 derlen=446
len=447 derlen=447
len=448 derlen=448
len=449 derlen=449
len=450 derlen=450
len=451 derlen=451
len=452 derlen=452
len=453 derlen=453
len=454 derlen=454
len=455 derlen=455
len=456 derlen=456
len=457 derlen=457
len=458 derlen=458
len=459 derlen=459
len=460 derlen=460
len=461 derlen=461
len=462 derlen=462
len=463 derlen=463
len=464 derlen=464
len=465 derlen=465
len=466 derlen=466
len=467 derlen=467
len=468 derlen=468
len=469 derlen=469
len=470 derlen=470
len=471 derlen=471
len=472 derlen=472
len=473 derlen=473
len=474 derlen=474
len=475 derlen=475
len=476 derlen=476
len=477 derlen=477
len=478 derlen=478
len=479 derlen=479
len=480 derlen=480
len=481 derlen=481
len=482 derlen=482
len=483 derlen=483
len=484 derlen=484
len=485 derlen=485
len=486 derlen=486
len=487 derlen=487
len=488 derlen=488
len=489 derlen=489
len=490 derlen=490
len=491 derlen=491
len=492 derlen=492
len=493 derlen=493
len=494 derlen=494
len=495 derlen=495
len=496 derlen=496
len=497 derlen=497
len=498 derlen=498
len=499 derlen=499
len=500 derlen=500
len=501 derlen=501
len=502 derlen=502
len=503 derlen=503
len=504 derlen=504
len=505 derlen=505
len=506 derlen=506
len=507 derlen=507
len=508 derlen=508
len=509 derlen=509
len=510 derlen=510
len=511 derlen=511
len=512 derlen=512
len=513 derlen=513
len=514 derlen=514
len=515 derlen=515
len=516 derlen=516
len=517 derlen=517
len=518 derlen=518
len=519 derlen=519
len=520 derlen=520
len=521 derlen=521
len=522 derlen=522
len=523 derlen=523
len=524 derlen=524
len=525 derlen=525
len=526 derlen=526
len=527 derlen=527
len=528 derlen=528
len=529 derlen=529
len=530 derlen=530
len=531 derlen=531
len=532 derlen=532
len=533 derlen=533
len=534 derlen=534
len=535 derlen=535
len=536 derlen=536
len=537 derlen=537
len=538 derlen=538
len=539 derlen=539
len=540 derlen=540
len=541 derlen=541
len=542 derlen=542
len=543 derlen=543
len=544 derlen=544
len=545 derlen=545
len=546 derlen=546
len=547 derlen=547
len=548 derlen=548
len=549 derlen=549
len=550 derlen=550
len=551 derlen=551
len=552 derlen=552
len=553 derlen=553
len=554 derlen=554
len=555 derlen=555
len=556 derlen=556
len=557 derlen=557
len=558 derlen=558
len=559 derlen=559
len=560 derlen=560
len=561 derlen=561
len=562 derlen=562
len=563 derlen=563
len=564 derlen=564
len=565 derlen=565
len=566 derlen=566
len=567 derlen=567
len=568 derlen=568
len=569 derlen=569
len=570 derlen=570
len=571 derlen=571
len=572 derlen=572
len=573 derlen=573
len=574 derlen=574
len=575 derlen=575
len=576 derlen=576
len=577 derlen=577
len=578 derlen=578
len=579 derlen=579
len=580 derlen=580
len=581 derlen=581
len=582 derlen=582
len=583 derlen=583
len=584 derlen=584
len=585 derlen=585
len=586 derlen=586
len=587 derlen=587
len=588 derlen=588
len=589 derlen=589
len=590 derlen=590
len=591 derlen=591
len=592 derlen=592
len=593 derlen=593
len=594 derlen=594
len=595 derlen=595
len=596 derlen=596
len=597 derlen=597
len=598 derlen=598
len=599 derlen=599
len=600 derlen=600
len=601 derlen=601
len=602 derlen=602
len=603 derlen=603
len=604 derlen=604
len=605 derlen=605
len=606 derlen=606
len=607 derlen=607
len=608 derlen=608
len=609 derlen=609
len=610 derlen=610
len=611 derlen=611
len=612 derlen=612
len=613 derlen=613
len=614 derlen=614
len=615 derlen=615
len=616 derlen=616
len=617 derlen=617
len=618 derlen=618
len=619 derlen=619
len=620 derlen=620
len=621 derlen=621
len=622 derlen=622
len=623 derlen=623
len=624 derlen=624
len=625 derlen=625
len=626 derlen=626
len=627 derlen=627
len=628 derlen=628
len=629 derlen=629
len=630 derlen=630
len=631 derlen=631
len=632 derlen=632
len=633 derlen=633
len=634 derlen=634
len=635 derlen=635
len=636 derlen=636
len=637 derlen=637
len=638 derlen=638
len=639 derlen=639
len=640 derlen=640
len=641 derlen=641
len=642 derlen=642
len=643 derlen=643
len=644 derlen=644
len=645 derlen=645
len=646 derlen=646
len=647 derlen=647
len=648 derlen=648
len=649 derlen=649
len=650 derlen=650
len=651 derlen=651
len=652 derlen=652
len=653 derlen=653
len=654 derlen=654
len=655 derlen=655
len=656 derlen=656
len=657 derlen=657
len=658 derlen=658
len=659 derlen=659
len=660 derlen=660
len=661 derlen=661
len=662 derlen=662
len=663 derlen=663
len=664 derlen=664
len=665 derlen=665
len=666 derlen=666
len=667 derlen=667
len=668 derlen=668
len=669 derlen=669
len=670 derlen=670
len=671 derlen=671
len=672 derlen=672
len=673 derlen=673
len=674 derlen=674
len=675 derlen=675
len=676 derlen=676
len=677 derlen=677
len=678 derlen=678
len=679 derlen=679
len=680 derlen=680
len=681 derlen=681
len=682 derlen=682
len=683 derlen=683
len=684 derlen=684
len=685 derlen=685
len=686 derlen=686
len=687 derlen=687
len=688 derlen=688
len=689 derlen=689
len=690 derlen=690
len=691 derlen=691
len=692 derlen=692
len=693 derlen=693
len=694 derlen=694
len=695 derlen=695
len=696 derlen=696
len=697 derlen=697
len=698 derlen=698
len=699 derlen=699
len=700 derlen=700
len=701 derlen=701
len=702 derlen=702
len=703 derlen=703
len=704 derlen=704
len=705 derlen=705
len=706 derlen=706
len=707 derlen=707
len=708 derlen=708
len=709 derlen=709
len=710 derlen=710
len=711 derlen=711
len=712 derlen=712
len=713 derlen=713
len=714 derlen=714
len=715 derlen=715
len=716 derlen=716
len=717 derlen=717
len=718 derlen=718
len=719 derlen=719
len=720 derlen=720
len=721 derlen=721
len=722 derlen=722
len=723 derlen=723
len=724 derlen=724
len=725 derlen=725
len=726 derlen=726
len=727 derlen=727
len=728 derlen=728
len=729 derlen=729
len=730 derlen=730
len=731 derlen=731
len=732 derlen=732
len=733 derlen=733
len=734 derlen=734
len=735 derlen=735
len=736 derlen=736
len=737 derlen=737
len=738 derlen=738
len=739 derlen=739
len=740 derlen=740
len=741 derlen=741
len=742 derlen=742
len=743 derlen=743
len=744 derlen=744
len=745 derlen=745
len=746 derlen=746
len=747 derlen=747
len=748 derlen=748
len=749 derlen=749
len=750 derlen=750
len=751 derlen=751
len=752 derlen=752
len=753 derlen=753
len=754 derlen=754
len=755 derlen=755
len=756 derlen=756
len=757 derlen=757
len=758 derlen=758
len=759 derlen=759
len=760 derlen=760
len=761 derlen=761
len=762 derlen=762
len=763 derlen=763
len=764 derlen=764
len=765 derlen=765
len=766 derlen=766
len=767 derlen=767
len=768 derlen=768
len=769 derlen=769
len=770 derlen=770
len=771 derlen=771
len=772 derlen=772
len=773 derlen=773
len=774 derlen=774
len=775 derlen=775
len=776 derlen=776
len=777 derlen=777
len=778 derlen=778
len=779 derlen=779
len=780 derlen=780
len=781 derlen=781
len=782 derlen=782
len=783 derlen=783
len=784 derlen=784
len=785 derlen=785
len=786 derlen=786
len=787 derlen=787
len=788 derlen=788
len=789 derlen=789
len=790 derlen=790
len=791 derlen=791
len=792 derlen=792
len=793 derlen=793
len=794 derlen=794
len=795 derlen=795
len=796 derlen=796
len=797 derlen=797
len=798 derlen=798
len=799 derlen=799
len=800 derlen=800
len=801 derlen=801
len=802 derlen=802
len=803 derlen=803
len=804 derlen=804
len=805 derlen=805
len=806 derlen=806
len=807 derlen=807
len=808 derlen=808
len=809 derlen=809
len=810 derlen=810
len=811 derlen=811
len=812 derlen=812
len=813 derlen=813
len=814 derlen=814
len=815 derlen=815
len=816 derlen=816
len=817 derlen=817
len=818 derlen=818
len=819 derlen=819
len=820 derlen=820
len=821 derlen=821
len=822 derlen=822
len=823 derlen=823
len=824 derlen=824
len=825 derlen=825
len=826 derlen=826
len=827 derlen=827
len=828 derlen=828
len=829 derlen=829
len=830 derlen=830
len=831 derlen=831
len=832 derlen=832
len=833 derlen=833
len=834 derlen=834
len=835 derlen=835
len=836 derlen=836
len=837 derlen=837
len=838 derlen=838
len=839 derlen=839
len=840 derlen=840
len=841 derlen=841
len=842 derlen=842
len=843 derlen=843
len=844 derlen=844
len=845 derlen=845
len=846 derlen=846
len=847 derlen=847
len=848 derlen=848
len=849 derlen=849
len=850 derlen=850
len=851 derlen=851
len=852 derlen=852
len=853 derlen=853
len=854 derlen=854
len=855 derlen=855
len=856 derlen=856
len=857 derlen=857
len=858 derlen=858
len=859 derlen=859
len=860 derlen=860
len=861 derlen=861
len=862 derlen=862
len=863 derlen=863
len=864 derlen=864
len=865 derlen=865
len=866 derlen=866
len=867 derlen=867
len=868 derlen=868
len=869 derlen=869
len=870 derlen=870
len=871 derlen=871
len=872 derlen=872
len=873 derlen=873
len=874 derlen=874
len=875 derlen=875
len=876 derlen=876
len=877 derlen=877
len=878 derlen=878
len=879 derlen=879
len=880 derlen=880
len=881 derlen=881
len=882 derlen=882
len=883 derlen=883
len=884 derlen=884
len=885 derlen=885
len=886 derlen=886
len=887 derlen=887
len=888 derlen=888
len=889 derlen=889
len=890 derlen=890
len=891 derlen=891
len=892 derlen=892
len=893 derlen=893
len=894 derlen=894
len=895 derlen=895
len=896 derlen=896
len=897 derlen=897
len=898 derlen=898
len=899 derlen=899
len=900 derlen=900
len=901 derlen=901
len=902 derlen=902
len=903 derlen=903
len=904 derlen=904
len=905 derlen=905
len=906 derlen=906
len=907 derlen=907
len=908 derlen=908
len=909 derlen=909
len=910 derlen=910
len=911 derlen=911
len=912 derlen=912
len=913 derlen=913
len=914 derlen=914
len=915 derlen=915
len=916 derlen=916
len=917 derlen=917
len=918 derlen=918
len=919 derlen=919
len=920 derlen=920
len=921 derlen=921
len=922 derlen=922
len=923 derlen=923
len=924 derlen=924
len=925 derlen=925
len=926 derlen=926
len=927 derlen=927
len=928 derlen=928
len=929 derlen=929
len=930 derlen=930
len=931 derlen=931
len=932 derlen=932
len=933 derlen=933
len=934 derlen=934
len=935 derlen=935
len=936 derlen=936
len=937 derlen=937
len=938 derlen=938
len=939 derlen=939
len=940 derlen=940
len=941 derlen=941
len=942 derlen=942
len=943 derlen=943
len=944 derlen=944
len=945 derlen=945
len=946 derlen=946
len=947 derlen=947
len=948 derlen=948
len=949 derlen=949
len=950 derlen=950
len=951 derlen=951
len=952 derlen=952
len=953 derlen=953
len=954 derlen=954
len=955 derlen=955
len=956 derlen=956
len=957 derlen=957
len=958 derlen=958
len=959 derlen=959
len=960 derlen=960
len=961 derlen=961
len=962 derlen=962
len=963 derlen=963
len=964 derlen=964
len=965 derlen=965
len=966 derlen=966
len=967 derlen=967
len=968 derlen=968
len=969 derlen=969
len=970 derlen=970
len=971 derlen=971
len=972 derlen=972
len=973 derlen=973
len=974 derlen=974
len=975 derlen=975
len=976 derlen=976
len=977 derlen=977
len=978 derlen=978
len=979 derlen=979
len=980 derlen=980
len=981 derlen=981
len=982 derlen=982
len=983 derlen=983
len=984 derlen=984
len=985 derlen=985
len=986 derlen=986
len=987 derlen=987
len=988 derlen=988
len=989 derlen=989
len=990 derlen=990
len=991 derlen=991
len=992 derlen=992
len=993 derlen=993
len=994 derlen=994
len=995 derlen=995
len=996 derlen=996
len=997 derlen=997
len=998 derlen=998
len=999 derlen=999
len=1000 derlen=1000
len=1001 derlen=1001
len=1002 derlen=1002
len=1003 derlen=1003
len=1004 derlen=1004
